query-planner-js/src/__tests__/buildPlan.test.ts

```diff
@@ -7973,3 +7973,195 @@ describe('@requires references external field indirectly', () => {
     `);
   });
 });
+
+describe('handles fragments with directive conditions', () => {
+  test('fragment with interseting parent type and directive condition', () => {
+    const subgraphA = {
+      typeDefs: gql`
+        directive @test on FRAGMENT_SPREAD
```

Comment:
Maybe a dumb question, but how do we restrict the types that the directive can be assigned to in the query? I assume the answer is "you can't"

Reply:
AFAIK you can only specify the locations so you cannot restrict it by types. Guessing you would need some external linter/validation logic to enforce it by type.